### PR TITLE
Use eccentricity to find graph centres

### LINKED ISSUE #2027: Graph center
**What is the feature or improvement you would like to see?**

A function that computed the [graph center](https://en.wikipedia.org/wiki/Graph_center), i.e the set of vertices with lowest eccentricity.

**Use cases for the feature**

This is a standard concept in graph theory.

**References**

 - https://en.wikipedia.org/wiki/Graph_center
 - `igraph_eccentricity()`

### PR DESCRIPTION
**Pull request by pradkrish**
Partially fixes #2027

### CODE REVIEW

File: src/paths/distances.c [hunk 1]
```diff
@@ -212,3 +212,34 @@ int igraph_radius(const igraph_t *graph, igraph_real_t *radius,
...
+
+    igraph_vector_t ecc;
+    igraph_vit_t vit;
+    igraph_vs_t vids = igraph_vss_all();
+
+    igraph_vector_init(&ecc, 0);
+
+    igraph_eccentricity(graph, &ecc, vids, mode);
+    IGRAPH_CHECK(igraph_vit_create(graph, vids, &vit));
+    if (!IGRAPH_VIT_END(vit)) {
```
Comment (szhorvat) [inline src/paths/distances.c:232]: This loop is unnecessary. We only need to find the vertices with maximum eccentricity, so we only need to loop over the eccentricity vector.

**Review by szhorvat — COMMENTED** [review]
I added a few preliminary comments, but I think it will be easier for you to open a new PR against `develop` instead of rebasing this one. It's up to you which way to proceed though.

**Comment by szhorvat** [comment]
FYI the test failures with MinGW are not related to your code, and have just been worked around `master` and `develop`. After you rebase this on `develop` (or open a new PR against `develop`), this annoyance will be gone.

File: src/paths/distances.c [hunk 2]
```diff
@@ -212,3 +212,34 @@ int igraph_radius(const igraph_t *graph, igraph_real_t *radius,
...
+}
+
+int igraph_center(const igraph_t *graph, igraph_vector_t *res,
+                  igraph_neimode_t mode) {
+
+    igraph_vector_t ecc;
+    igraph_vit_t vit;
+    igraph_vs_t vids = igraph_vss_all();
+
+    igraph_vector_init(&ecc, 0);
```
[the remark this answers is not in the document]
Comment (pradkrish) [reply src/paths/distances.c:228]: Ack

File: src/paths/distances.c (hunk 1 above)
Comment (pradkrish) [reply src/paths/distances.c:232]: The idea of adding the condition `if (!IGRAPH_VIT_END(vit))` is to check for Null graph. If this is not correct, is there another way of doing it?

File: src/paths/distances.c [hunk 3]
```diff
@@ -212,3 +212,34 @@ int igraph_radius(const igraph_t *graph, igraph_real_t *radius,
 
     return 0;
 }
+
+long int min(long int a, long int b) 
+{
+    return (a > b) ? b : a;
+}
+
+int igraph_center(const igraph_t *graph, igraph_vector_t *res,
```
[the remark this answers is not in the document]
Comment (pradkrish) [reply src/paths/distances.c:221]: Ack

**Comment by pradkrish** [comment]
> FYI the test failures with MinGW are not related to your code, and have just been worked around `master` and `develop`. After you rebase this on `develop` (or open a new PR against `develop`), this annoyance will be gone.

Yeah, it was an oversight on my part, I thought I was on develop branch. Also, it looks like the unit tests I have added is not building as part of the test suite.. I have added it to `tests/CMakeList.txt`. Am I missing something here?

File: src/paths/distances.c (hunk 1 above)
Comment (szhorvat) [reply src/paths/distances.c:232]: Good point that we must check for the null graph, otherwise `vector_min` fails. Do this by checking the vertex count.

**Comment by pradkrish** [comment]
That doesn't look good at all, best to abandon this and open a new PR.

**Pull request closed without merging**